inference-engine/src/hetero_plugin/hetero_executable_network.cpp

```diff
@@ -312,14 +347,377 @@ HeteroExecutableNetwork::HeteroExecutableNetwork(const InferenceEngine::ICNNNetw
         auto deviceName = d._device;
         auto metaDevices = _heteroPlugin->GetDevicePlugins(deviceName, cfg);
         assert(metaDevices.size() == 1);
-
         auto loadConfig = metaDevices[deviceName];
         d._network = _heteroPlugin->GetCore()->LoadNetwork(d._clonedNetwork, deviceName, loadConfig);
     }
 
     networks = std::move(descs);
 }
 
+template<typename T>
+using NodeMap = std::unordered_map<ngraph::Node*, T>;
+
+void HeteroExecutableNetwork::InitNgraph(const InferenceEngine::ICNNNetwork& network_) {
+    auto function = network_.getFunction();
+    auto networkPtr = cloneNetwork(network_);
+    auto& network = *networkPtr;
+    auto itDumpDotFile = _config.find(HETERO_CONFIG_KEY(DUMP_GRAPH_DOT));
+    bool dumpDotFile = itDumpDotFile != _config.end() ? (itDumpDotFile->second == YES) : false;
+#ifndef NDEBUG
+    dumpDotFile  = true;
+#endif
+
+    QueryNetworkResult queryNetworkResult;
+    auto orderedOps = function->get_ordered_ops();
+    orderedOps.erase(
+        std::remove_if(std::begin(orderedOps), std::end(orderedOps), [] (const std::shared_ptr<ngraph::Node>& node) {
+            return node->is_constant();
+        }),
+        std::end(orderedOps));
+    bool allEmpty = true;
+    for (auto&& node : orderedOps) {
+        auto& nodeInfo = node->get_rt_info();
+        auto itInfo = nodeInfo.find("affinity");
+        if (itInfo != nodeInfo.end()) {
+            queryNetworkResult.supportedLayersMap.emplace(
+                node->get_friendly_name(),
+                Parameter{itInfo->second}.as<std::string>());
```

Comment:
We are not using ie Parameter for rt info attributes. May be it's better to use variant directly?

Reply:
Resently I was adwised to use ie::Parameter Here. @ilyachur @ilya-lavrenov what can you say?

Reply:
@apankratovantonp This mechanism was refactored and all at the current moment transformations library doesn't contain any dependencies on IE. Moreover the rt propagation mechanism doesn't work with IE Parameter. @GlebKazantaev is right. Please change it to nGraph Variant.

Reply:
Done